src/saveload/town_sl.cpp

```diff
@@ -80,22 +80,22 @@ void UpdateHousesAndTowns()
 		if (!IsTileType(t, MP_HOUSE)) continue;
 
 		HouseID house_type = GetCleanHouseType(t);
-		TileIndex north_tile = t + GetHouseNorthPart(house_type); // modifies 'house_type'!
+		TileIndex north_tile = static_cast<TileIndex>(t) + GetHouseNorthPart(house_type); // modifies 'house_type'!
 		if (t == north_tile) {
 			const HouseSpec *hs = HouseSpec::Get(house_type);
 			bool valid_house = true;
 			if (hs->building_flags & TILE_SIZE_2x1) {
-				TileIndex tile = t + TileDiffXY(1, 0);
+				TileIndex tile = static_cast<TileIndex>(t) + TileDiffXY(1, 0);
 				if (!IsTileType(tile, MP_HOUSE) || GetCleanHouseType(tile) != house_type + 1) valid_house = false;
 			} else if (hs->building_flags & TILE_SIZE_1x2) {
-				TileIndex tile = t + TileDiffXY(0, 1);
+				TileIndex tile = static_cast<TileIndex>(t) + TileDiffXY(0, 1);
 				if (!IsTileType(tile, MP_HOUSE) || GetCleanHouseType(tile) != house_type + 1) valid_house = false;
 			} else if (hs->building_flags & TILE_SIZE_2x2) {
-				TileIndex tile = t + TileDiffXY(0, 1);
+				TileIndex tile = static_cast<TileIndex>(t) + TileDiffXY(0, 1);
 				if (!IsTileType(tile, MP_HOUSE) || GetCleanHouseType(tile) != house_type + 1) valid_house = false;
-				tile = t + TileDiffXY(1, 0);
+				tile = static_cast<TileIndex>(t) + TileDiffXY(1, 0);
 				if (!IsTileType(tile, MP_HOUSE) || GetCleanHouseType(tile) != house_type + 2) valid_house = false;
-				tile = t + TileDiffXY(1, 1);
+				tile = static_cast<TileIndex>(t) + TileDiffXY(1, 1);
 				if (!IsTileType(tile, MP_HOUSE) || GetCleanHouseType(tile) != house_type + 3) valid_house = false;
```

Comment:
Would using `for (const TileIndex t : Map::Iterate()) {` avoid all this casting, or will that no longer work as well?